[T12-2] Acquizition

sergiovieri opened the pull request:
@robincyu 
@lucatriglia 
@lungsin

lucatriglia committed:
Update AboutUs and ContactUs info (#1)

* docs: update AboutUs info

* docs: update ContactUs info

* docs/AboutUs: fix image name

* docs: add project portfolio page for each member

Commit:
Update Readme badges (#3)

* readme: update travis and codacy badges

* readme: update coveralls badge

bqnguyen94 commented on README.adoc:
```diff
@@ -1,10 +1,9 @@
 = Address Book (Level 4)
```
Please replace `Address Book` with your product's name everywhere in the documents as well

bqnguyen94 commented on README.adoc:
```diff
@@ -1,10 +1,9 @@
+https://www.codacy.com/app/sergiovieri/main?utm_source=github.com&utm_medium=referral&utm_content=cs2103-ay1819s2-t12-2/main&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/caf9760ff67c48b498d39589c74ac45a[Codacy Badge]]
 
 ifdef::env-github[]
 image::docs/images/Ui.png[width="600"]
```
Don't forget to update this picture with a mockup or actual GUI of your product, and add `se-edu` to `Acknowledgements` below

bqnguyen94 reviewed (commented):
I see that you have set up the repository and started working on your project 👍

lucatriglia committed:
Update About us (#10)

* docs: update AboutUs info

* docs: update ContactUs info

* docs/AboutUs: fix image name

* docs: add project portfolio page for each member

* AboutUs: fix portfolio links

sergiovieri committed:
Update Readme (#4)

* readme: add appveyor badge

* README.adoc: Change app name to Acquizition

* Update README.adoc

* Update UI image

* Update README.adoc to include reference to se-edu

lungsin committed:
Update user guide (#6)

* add Introduction and description for add command

* change docs for list command

* docs/UserGuide.adoc: edit command

* docs/UserGuide.adoc: fix format + docs for find command

* docs/UserGuide.adoc: add docs for delete command

* docs/UserGuide.adoc: add coming in future work for undo, redo, select, history

* docs/UserGuide.adoc: add docs for quiz mode

* docs/UserGuide.adoc: add docs for stats

* docs/UserGuide.adoc: make 'coming soon' tag shorter

* docs/UserGuide.adoc: remove select command, restructure the order of undo and redo

* docs/UserGuide.adoc: refactor  to

* docs/UserGuide.adoc: fix typos and edit quick start

* docs/UserGuide.adoc: fix typos and change  section

* docs/UserGuide.adoc: change features section to be more consistent with other commands

* docs/UserGuide.adoc: change command summary

lucatriglia committed:
[WIP] Add Upload Functionality  (#12)

* Add upload command

* docs/UserGuide.adoc: update user guide to include upload command

* docs/ContactUs.adoc: fix link for issue tracker

* docs/UserGuide: include information about the user guide

* commands/UploadCommand: add newline for message usage

sergiovieri committed:
Add simple quiz mode (#8)

* Add Quiz command class

* Fix checkstyle errors

* Change browser panel to card view panel

* Random quiz card view

* test: remove broswer from test

* Fix styles

* test: Comment out close help test

* Remove browserPanel test

lungsin committed:
Add use cases for developer guide (#16)

* Docs: add use cases for edit, add, delete

* docs: add use cases for quiz mode

* docs: add use cases for find

* docs: add use cases for import and export cards

* change asciidoc attribute

* change team name from SE-EDU to T12-2

lucatriglia committed:
docs/DeveloperGuide: update user stories (#17)

sergiovieri committed:
readme: Update travis repository name (#18)

robincyu committed:
Refactor AddressBook to CardCollection, Person to Flashcard (#14)

* Refactor AddressBook to CardCollection

* Fix code style

* Refactor Person to Flashcard

* Continue refactoring

* Fix code style

* Fix minor codestyle errors

* LICENSE: Revert flashcard -> person

sergiovieri committed:
README: Update badge repository name (#23)

sergiovieri committed:
Refactor Flashcard Fields (#22)

* Add Face class

* Fix fxml imports

* Remove browser panel code and tests

* Change flash card fields to front face and back face

* Complete EditCommandTest

* Complete EditCommandSystemTest

* Complete CardCollectionTest

* Complete JsonAdapted tests

* Complete ModelManagerTest

* Complete VersionedCardCollectionTest

* Finish all tests

* Fix imports

* Remove all unused Flashcard field classes

lungsin committed:
Add statistics functionality (#13)

* Logic: teach the app to accept 'stats' command which does nothing

* Refactor: getTypicalAddressBook -> getTypicalCardCollection

* Add Statistics class and test

* StatisticsTest: add test on merge method

* Model: add json validation in Person, add field Statistics in Person

* Logic: Statistics functionality

Currently, command `stats` will display the success rate of quiz mode of
all flashcards.

* Test: change the placeholder test for StatsCommand

* Storage: add a field in JSON + adapt the test accordingly

sergiovieri committed:
Modify Flashcard Card View (#27)

* Refactor FlashcardCard to FlashcardListCard to avoid ambiguity

* Add FlashcardCardView and change the CardViewPanel to use it

* Add FlashcardCardView tests

* Modify card view and remove primary stage decoration

* Modify card view to use separate panes

lucatriglia committed:
Enhance Find Command (#29)

* Enhance Find Command

* model/flashcard: Add KeywordsPredicate

* logic/parser: Update Find Command Parsing

* logic/commands: Update Commands for Find

* docs/UserGuide: Update UG to include enhanced find

* Update tests for enhanced find command

* Update Developer Guide to include find command implementation

sergiovieri committed:
Add quiz command to developer guide (#30)

* Add quiz command to developer guide

* DeveloperGuide: Add quiz mode design considerations

lungsin committed:
Add Statistics section in developer guide (#36)

* Add Statistics section in developer guide

robincyu committed:
Add Image Support to DeveloperGuide.adoc (#38)

sergiovieri committed:
Reposense: Add config.json (#39)

lungsin committed:
Implement Filtered statistics (#40)

* Logic: add filter functionality to statsCommand

* Test: add test for StatsCommandParser

* FindCommand.java: Fix MESSAGE_USAGE

* Tests: add test for stats command with predicate

* Bug Fix: fix equals method of FlashcardContainsKeywordsPredicate

* Test: add more tests

* Tests: add stats command parser test and TODO

lungsin committed:
Rename _reponense -> _reposense (#42)

sergiovieri committed:
Advanced Quiz Mode (#46)

- Quiz mode that sorts cards by statistics

- Quiz panel that contains cards remaining, number of good and bad feedback

- User feedback through GoodCommand, BadCommand

- ShowCommand to show the back face of a card during quiz mode

- SmartCommand that executes ShowCommand if back face is not yet shown, or GoodCommand if back face is already shown

* Add quiz mode property to relevant places

* ExitCommand: If now is quiz mode, exit from quiz mode.
Otherwise, exit as normal.

* Hide flashcard list view on quiz mode

* Add user feedback on quiz mode, and quiz mode now shuffles cards

* Add quiz mode panel

* Add show command and modify some quiz behaviours

* Add smart command

* QuizPanel: Add good/bad button

* Quiz: Add current quiz statistics

* Add all new class tests

* QuizMode: Update statistics on user feedback

* Quiz: Commit model on user feedback

* Bump version to 1.3

* Fix selection bugs and inconsistencies

* Quiz: Refactor magic constants to QuizState

lucatriglia committed:
Share command Implementation  (#43)

* Add share command implementation

* FlashcardContainsKeywordsPredicate: Fix equals method bug

* tests: Test find command Implementation and parser

* docs/UserGuide.adoc: update UG to include share command

* Update Share Command

sergiovieri committed:
Add Toolbar (#49)

* Add toolbar

* Fix style

* Add acquizition label to toolbar and adjust various sizes

* Add acquizition icon

* Change cards remaining label when 1 remaining card

* Add toolbar tests

* QuizPanel: Add JavaDoc

* Disallow non-quiz commands in quiz mode

* AddCommandTest: Support quiz mode

* Add UploadCommandTest and ShareCommandTest

* ShareCommand: Fix path resolution

robincyu committed:
Image support (#50)

* Add placeholder image support

* Add proper image support

* Allow image removal by specifying empty i/ tag

* Add image command which allows importing of images into Acquizition

* Add tests for ImageCommand

* Add required space in Messages.java

* FlashcardCardView: Restore shade

lucatriglia committed:
Import Command  (#53)

* Update upload command to import command

sergiovieri committed:
UserGuide: Update quiz mode (#52)

* UserGuide: Update quiz mode

* UserGuide: Add good and bad command

* UserGuide: Add show command

* UserGuide: Add smart command

* Add quiz ui image

* Add ui quiz screenshot and fix card view color

* UserGuide: Add quiz ui

* UserGuide: Fix unclosed tag

robincyu committed:
Update User Guide to v1.3 (#54)

* UserGuide: Update quiz mode

* UserGuide: Add good and bad command

* UserGuide: Add show command

* UserGuide: Add smart command

* Add quiz ui image

* Add ui quiz screenshot and fix card view color

* UserGuide: Add quiz ui

* UserGuide: Fix unclosed tag

* Update User Guide

* Add newline to UG

sergiovieri committed:
ImageCommand Fix (#56)

* ImageCommand: Fix image directory creation

* FlashcardCardView: Fix overflow ellipsis to wrap text

sergiovieri committed:
UserGuide: Add missing commands and command summary (#57)

* UserGuide: Add select command

* UserGuide: Add list command

* UserGuide: Add history command

* UserGuide: Add undo and redo commands

* UserGuide: Add command summary

* UserGuide: Add redo command

* UserGuide: Add quiz and command summary tags

lungsin committed:
Refactor filter (#81)

* Refactor: move filter part of FindCommandParser and StatsCommandParser to ParserUtil::filterByKeyword

* Logic: filter backface or frontface by substring instead of word matching

* Logic: list on empty flashcards will display other message

* Tests: modify test, command is not case sensitive

sergiovieri committed:
Change app icon (#87)

sergiovieri committed:
Developer Guide: Refactor existing commands and diagrams (#89)

* Modify architecture diagram

* Modify some class diagrams

* Remove encryption and add quiz sequence diagram pptx

* Modify undo redo diagrams

* Remove forking notes

lungsin committed:
Spaced Repetition System and Review mode (#86)

* Model: add proficiency

* Storage: integrate with flashcard

* Model: Proficiency now has inactive period field

* Model: Review mode in quiz

* UI: add is review mode in the UI

* Logic: add SRS mode to quiz

* Logic: Review mode will still update the statistics

* Test: add unit test for QuizCommand and its parser

* Test: add system tests for QuizCommand

* Epic: History does not accept empty string, Undo Redo provide command description

* Storage: After quiz is done, commit the result

* Make statistics and proficiency final

* Bug: click on flashcard list will print directly to resultBox

* Tests: Proficiency and statistics tests

* UI: Select with clicking

* Fix comand history bug and change theme

robincyu committed:
Image support fixes (#92)

* Fix issues related to image support and other minor grammar issues with other commands

* Fix Image Command showing full path instead of file name

* Fix black background color of list view when filtered

lucatriglia committed:
Update FindCommand to filter by success rate (#85)

* Logic: Update FindCommand to filter by success rate

* Tests: Add tests for find, import, share command

* Docs: Update UG and DG

* Docs: Add PPP

* Docs: Update lucatriglia.adoc

sergiovieri committed:
DeveloperGuide: Quiz Mode (#93)

* Remove hand cursor from good in quiz panel

* Add quiz, show, good/bad, exit diagrams and explanations

* Change design considerations to use bullet points

lungsin committed:
Selecting flashcard display date until its use in quiz (#91)

* Epic: now selecting card can see the time left until activation

* Test: add one more case

* Logic: change message to 'in quiz status'

robincyu committed:
Complete Rewrite of User Guide: Robin's and Neutral Features (#95)

* Comprehensive User Guide edits

* More changes to User Guide

sergiovieri committed:
Docs: Add PPP - Sergio Vieri (#59)

* Docs: Add sergio vieri PPP

* Fix exit quiz sequence diagram arrow color

* Fix user guide typo, add link to reposense in PPP, and replace old screenshots

* Add image and intro to PPP

robincyu committed:
Developer Guide: Robin's Features (#97)

* Developer guide changes

* Add tags to User Guide

* Fix quiz User Guide

sergiovieri committed:
Remove use cases from PPP (#98)

lungsin committed:
Docs srs stats (#94)

* Docs: developer guide undo redo

* add Proficiency in model class diagram

* docs: dev guide for SRS and Stats

* Docs: add diagram for StatsCommand

sergiovieri committed:
Quiz Fix (#100)

Quiz Fix

robincyu committed:
Add Robin's PPP (#99)

lucatriglia committed:
Update AboutUs.adoc (#101)

* Update AboutUs.adoc

* Delete lucatriglia.png
